src/OrbitBase/ProcessIdLinux.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <filesystem>
+#include <optional>
+#include <system_error>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ProcessId.h"
+#include "OrbitBase/Result.h"
+#include "absl/strings/numbers.h"
+
+namespace orbit_base {
+
+namespace fs = std::filesystem;
+
+namespace {
+
+std::optional<pid_t> ProcEntryToPid(const std::filesystem::directory_entry& entry) {
+  if (!entry.is_directory()) {
+    return std::nullopt;
+  }
+
+  int potential_pid;
+  if (!absl::SimpleAtoi(entry.path().filename().string(), &potential_pid)) {
+    return std::nullopt;
+  }
+
+  if (potential_pid <= 0) {
+    return std::nullopt;
+  }
+
+  return static_cast<pid_t>(potential_pid);
+}
+
+}  // namespace
+
+std::vector<pid_t> GetAllPids() {
+  fs::directory_iterator proc{"/proc"};
+  std::vector<pid_t> pids;
+
+  for (const auto& entry : proc) {
+    if (auto pid = ProcEntryToPid(entry); pid.has_value()) {
```

Comment:
Move fist part out of if?

Reply:
This is actually something that in other reviews we agreed to do: when the variable is only used in the if block, it's better to scope it and have it in the condition (imagine there was other code after the if).

Reply:
Agreed, looks weird.